app/components/ManageCart/reducers/reducer.js

```diff
@@ -43,11 +43,10 @@ const reducer = (state = initialState, action) => {
     return newState
 
   case REMOVE_ITEM:
-    const index = newState.cartItems.findIndex(action.item)
-    newState.cartItems.slice(index, 1)
+    newState.currentCart.products = state.currentCart.products.filter(prod => prod !== action.item)
```

Comment:
If the items have IDs, it is better to filter by ID than object equality. Imagine you get a response from the backend, the object created won't be the same object you already have in memory.